qute.ls/com.redhat.qute.ls/src/main/java/com/redhat/qute/settings/FaultTolerantTypeAdapterFactory.java

```diff
@@ -0,0 +1,69 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package com.redhat.qute.settings;
+
+import java.io.IOException;
+import java.util.logging.Logger;
+
+import com.google.gson.Gson;
+import com.google.gson.TypeAdapter;
+import com.google.gson.TypeAdapterFactory;
+import com.google.gson.reflect.TypeToken;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+/**
+ * Creates TypeAdapters that are wrapped in a try/catch statement, so if
+ * anything goes wrong when deserializing an object, it is set to null and not
+ * parsed.
+ *
+ */
+public class FaultTolerantTypeAdapterFactory implements TypeAdapterFactory {
```

Comment:
Do we still want the name to be "FaultTolerant"? I know we use this exact same class in lemminx, but fault tolerance is related to mp correct?

Reply:
No its not linked to lemminx or mp. It means that the load is fault tolerant.